### PR TITLE
Feature/Add Order History with Filters (fromDate, toDate, sale_status, payment_status, orderById)

### PR DESCRIPTION
**Pull request by SyedSamrozeAli**

### CODE REVIEW

File: app/Http/Controllers/OrderController.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+                    $orders = $query->get();
+
+                } else {
+                    // sending error response means we didn't find data for given filters 
+                    return errorResponse("No orders fetched", 404);
+                }
+            }
+
+            return response()->json(ResponseHelper::sendResponse(true, 200, "Data Fetched Successfully", Order_Resource::collection($orders), $request->paginate, $request->pageSize, $request->pageNo));
```
Comment (ZeeshanHabib7) [inline app/Http/Controllers/OrderController.php]: please maintain same response over the application

File: app/Http/Requests/OrderRequest.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, \Illuminate\Contracts\Validation\ValidationRule|array<mixed>|string>
+     */
+    public function rules(): array
+    {
+        return [
+            'orderById' => 'string|in:true,false',
+            'fromDate' => 'date',
```
Comment (ZeeshanHabib7) [inline app/Http/Requests/OrderRequest.php]: validate the date format as well

File: app/Models/Products.php [hunk 3]
```diff
@@ -2,26 +2,47 @@
...
 namespace App\Models;
 
-use App\Http\Requests\ProductRequest_SA;
+use App\Http\Requests\ProductRequest;
 use Illuminate\Database\Eloquent\Factories\HasFactory;
 use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\SoftDeletes;
 
-class Products_SA extends Model
+class Products extends Model
```
Comment (hfatima08) [inline app/Models/Products.php]: Model Name should be singular

File: app/Http/Requests/OrderRequest.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+     *
+     * @return array<string, \Illuminate\Contracts\Validation\ValidationRule|array<mixed>|string>
+     */
+    public function rules(): array
+    {
+        return [
+            'orderById' => 'string|in:true,false',
+            'fromDate' => 'date|date_format:Y-m-d',
+            'toDate' => 'date|date_format:Y-m-d',
+            'payment_status' => 'string|in:paid,unpaid',
```
Comment (hfatima08) [inline app/Http/Requests/OrderRequest.php:30]: We have changed the payment_status column definition to  $table->string('payment_status')->nullable()->change();
It's not an enum of paid & unpaid anymore.

Comment (SyedSamrozeAli) [reply app/Http/Requests/OrderRequest.php:30]: But values for the payment status should be predefined. Like we have done in order_status (order_placed,order_rejected...).

**Pull request merged**